src/components/Article.jsx

```diff
@@ -1,28 +1,106 @@
-import { getArticle } from "../../api";
+import { getArticle, patchArticle } from "../../api";
 import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import moment from "moment";
 import Comments from "./Comments";
+import { GrLike, GrDislike } from "react-icons/gr";
 
 const Article = () => {
   const { article_id } = useParams();
   const [article, setArticle] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [votes, setVotes] = useState(0);
+  const [myVote, setMyVote] = useState(null);
+  const [buttonDislikeHighlighted, setButtonDislikeHighlighted] =
+    useState(null);
+  const [buttonLikeHighlighted, setButtonLikeHighlighted] = useState(null);
```

Comment:
I'm certain you could reduce how much state you're changing here, does every part of state here need to be independent or could it be accessed if put together, for example the button highlights, you could store both booleans in an object and have the interaction be the same.